src/mjolnir/shortcutbuilder.cc

```diff
@@ -77,17 +90,10 @@ bool EdgesMatch(const graph_tile_ptr& tile, const DirectedEdge* edge1, const Dir
     return false;
   }
 
-  // Names must match
-  // TODO - this allows matches in any order. Do we need to maintain order?
-  // TODO - should allow near matches?
-  std::vector<std::string> edge1names = tile->GetNames(edge1);
-  std::vector<std::string> edge2names = tile->GetNames(edge2);
-  std::sort(edge1names.begin(), edge1names.end());
-  std::sort(edge2names.begin(), edge2names.end());
-  if (edge1names != edge2names)
-    return false;
-
   // if they have access restrictions those must match (for modes that use shortcuts)
+  // TODO(nils): same as above: same as access masks
+  //   IMO, we should take the most restrictive value for the entire shortcut
+  //   will be a bit harder to determine for timed restrictions though
```

Comment:
one more comment here: access restrictions are mode specific. but that doesn't matter for most as those modes are "static" (always the same, e.g. height/weight etc). so it still makes sense to only copy the most restrictive one for each type.

only for conditional restrictions modes are dynamic. to not make our life too hard here, I'd propose to simply copy all of the conditional ones onto the shortcut. there's few ones anyways, even in commercial datasets, and if we'd break up shortcuts because of them we'd look at them anyways.